java/src/main/java/ai/rapids/cudf/HostColumnVector.java

```diff
@@ -929,74 +929,127 @@ public ColumnBuilder appendStructValues(StructData... inputList) {
     }
 
     /**
-     * A method that is responsible for growing the buffers as needed
-     * and incrementing the row counts when we append values or nulls.
-     * @param hasNull indicates whether the validity buffer needs to be considered, as the
-     *                nullcount may not have been fully calculated yet
-     * @param length used for strings
+     * Grows valid buffer lazily. The valid buffer won't be materialized until the first null
+     * value appended. This method reuses the rowCapacity to track the sizes of column.
+     * Therefore, please call specific growBuffer method to update rowCapacity before calling
+     * this method.
+     */
+    private void growValidBuffer() {
```

Comment:
It looks like the validity buffer only grows when we append a null.  But, HostColumnVectorCore that takes the buffer over after the builder is done thinks that it got a buffer that is large enough to hold validity for all of the rows in the column, or an empty buffer if there are no nulls. If we have assertions enabled, then we will get errors trying to access the data or copy it to the device. If they are not enabled, I am scared we are going to copy data off the end of the buffers and cause some real issues.


I think we either have to grow the validity in sync with the `rowCapacity`, or we have to have a step before we build the HostColumnVector* and check that the validity is large enough to hold all of the rows.